Kernel/Devices/HID/PS2/KeyboardDevice.cpp

```diff
@@ -31,35 +29,27 @@ void PS2KeyboardDevice::irq_handle_byte_read(u8 byte)
     event.e0_prefix = m_has_e0_prefix;
     m_has_e0_prefix = false;
 
-    dbgln_if(KEYBOARD_DEBUG, "Keyboard::irq_handle_byte_read: {:#02x} {}", ch, (pressed ? "down" : "up"));
+    dbgln_if(KEYBOARD_DEBUG, "Keyboard::handle_byte_read_from_serial_input: {:#02x} {}", ch, (pressed ? "down" : "up"));
     event.scan_code_value = ch;
     m_keyboard_device->handle_scan_code_input_event(event);
 }
 
-bool PS2KeyboardDevice::handle_irq(RegisterState const&)
-{
-    // The controller will read the data and call irq_handle_byte_read
-    // for the appropriate device
-    return m_i8042_controller->irq_process_input_buffer(instrument_type());
-}
-
-UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<PS2KeyboardDevice>> PS2KeyboardDevice::try_to_initialize(I8042Controller const& ps2_controller, KeyboardDevice const& keyboard_device)
+UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<PS2KeyboardDevice>> PS2KeyboardDevice::try_to_initialize(PS2Controller const& ps2_controller, PS2PortIndex port_index, KeyboardDevice const& keyboard_device)
```

Comment:
Is the logic for this Keyboard device actually 100% removed from the I8042 chipset?

Reply:
There's not much logic to initialize a keyboard device (also no really specific logic in i8042), so yes, this is quite detached. The translation thing AFAIK is something that only the i8042 cared to implement, but technically other (unknown to me) controllers could use a translation port, so that's fine too.
In #18342 I give this a bit more thought, but this is OK for now too. :)